### DISCUSSION TITLE
Consider how rewrite could leverage the language server protocol for IDE integration

### DISCUSSION THREAD

**Discussion by aegershman** [openrewrite/rewrite#236] [Ideas]
This is a hypothetical. This isn't intended as something "pressing" or a direct-action item. Just externalizing the thought.

In contrast to writing individual "heavy" IDE plugins (vscode, intellij, eclipse, etc.)-- assuming IDE plugins would even be used-- consider what a `rewrite` ["language server"](https://microsoft.github.io/language-server-protocol/) could look like.

**Comment by reckart** [comment]
It seems that the Sprint Tool Suite 4 has support for OpenRewrite through a language server. See:

* https://entwickler.de/spring/spring-boot-update-spring-tools-001
* https://www.eclipsecon.org/2022/sessions/automate-code-changes-bringing-openrewrite-ide

Does anybody have experience with that? In particular, I would like to know if it may be possible to use this without having to install the entire STS.

**Reply by timtebeek** [reply]
We indeed have calls every two to three weeks with the team behind https://github.com/spring-projects/sts4 ; you might want to ask them what's possible with what they have delivered already, especially now that [IntelliJ is also starting to support LSP](https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/).

**Reply by anbusampath** [reply]
https://www.youtube.com/watch?v=hp5G-k8L1Xo

**Comment by timtebeek** [comment]
Closing as stale; feel free to open a new discussion if this is still relevant.
